include/aws/mqtt/client.h

```diff
@@ -422,6 +443,23 @@ int aws_mqtt_client_connection_set_reconnect_timeout(
     uint64_t min_timeout,
     uint64_t max_timeout);
 
+/**
+ * Sets the callbacks to call when a connection is success
```

Comment:
Since we're setting both success and failure here, we should probably change this to something like "Sets the callbacks to call when a connection succeeds or fails"